privateSBOMExchange/src/petra/lib/models/__init__.py

```diff
@@ -646,3 +696,147 @@ def build_sbom_tree(parser:SBOMParser):
     return root
 
 
+class GetTargetNodes:
+    """Visitor that collects the data and hash of each node.
+    This is used to test for membership"""
+
+    def __init__(self):
+        # Initialize a list to store the hashes of visited nodes
+        self.hashes = []
+
+    def visit_field_node(self, node):
+        # Append the hash if it exists, otherwise raise error
+        try:
+            node_hash = node.hash.hex()
+            self.hashes.append(node_hash)
+        except AttributeError:
+            pass
+    def visit_complex_node(self, node):
+        # Add the complex node's hash, if available
+        try:
+            node_hash = node.hash.hex()
+            self.hashes.append(node_hash)
+        except AttributeError:
+            pass
+
+        # Visit each child node
+        for child in node.children:
+            child.accept(self)
+
+    def visit_sbom_node(self, node):
+        # Add the SBOM node's hash, if available, and visit children
+        try:
+            node_hash = node.hash.hex()
+            self.hashes.append(node_hash)
+        except AttributeError:
+            pass
+        # Visit each child node
+        for child in node.children:
+            child.accept(self)
+
+    def get_hashes(self):
+        """
+        Returns the collected hashes to the caller.
+        Called after the tree traversal is complete.
+        """
+        return self.hashes
+
+
+def get_membership_proof(root, target_hash):
```

Comment:
Please move `get_membership_proof()` and `verify_membership_proof()` to the models/tree_ops.py file.